File: mobile/src/main/java/com/keepnotes/storage/GsonWrapper.java
package com.keepnotes.storage;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

public final class GsonWrapper {
    private static Gson gson;

    synchronized public static Gson getGson() {
        if (gson == null) {
            gson = new GsonBuilder().create();
        }
        return gson;
    }
}

File: mobile/src/main/java/com/keepnotes/storage/GsonWrapper.kt
package com.keepnotes.storage

import com.google.gson.Gson
import com.google.gson.GsonBuilder

object GsonWrapper {
    private var gson: Gson? = null

    @Synchronized fun getGson(): Gson {
        if (gson == null) {
            gson = GsonBuilder().create()
        }
        return gson ?: throw AssertionError("Set to null by another thread")
    }

}
